kgforge/specializations/models/rdf/rdf_model_service_from_store.py

```diff
@@ -59,24 +61,22 @@ def materialize(self, iri: URIRef) -> NodeProperties:
 
     def _validate(self, iri: str, data_graph: Graph) -> Tuple[bool, Graph, str]:
         # _type_shape will make sure all the shapes for this type are in the graph
-        self._type_shape(iri)
+        self._load_and_get_type_shape(URIRef(iri))
         return validate(data_graph, shacl_graph=self._graph)
 
     def resolve_context(self, iri: str) -> Dict:
-        if iri in self._context_cache:
-            return self._context_cache[iri]
-        document = self.recursive_resolve(iri)
-        self._context_cache.update({iri: document})
-        return document
+        if iri not in self._context_cache:
+            self._context_cache[iri] = self.recursive_resolve(iri)
+
+        return self._context_cache[iri]
 
     def generate_context(self) -> Dict:
-        for v in self._shapes_to_resources.values():
-            self._load_shape(v)
-        # reloads the shapes graph
-        self._sg = ShapesGraphWrapper(self._graph)
+        for v in self.schema_to_source.values():
+            self._load_shape_and_reload_shapes_graph(v)
```

Comment:
I think one thing doing one thing is better than _load_shape_and_reload_shapes_graph

Reply:
The _graph and the _shapes_graph are supposed to stay consistent so it's better to always _load_shape and put it in the shapes_graph automatically, rather than expect the person calling _load_shape to keep them consistent